rts/Sim/Units/CommandAI/CommandAI.cpp

```diff
@@ -1212,6 +1212,49 @@ void CCommandAI::ExecuteInsert(const Command& c, bool fromSynced)
 	SlowUpdate();
 }
 
+const std::optional<int> CCommandAI::FindTagIndex(const CCommandQueue& queue, unsigned int tag) const
+{
+	int i = 0;
+	for (const auto& qc: queue) {
+		if (qc.GetTag() == tag)
+			return i;
+		++i;
+	}
+	return std::nullopt;
+}
+
+
+const std::optional<std::pair<int, int>> CCommandAI::GetRemoveLimitsFromOptions(const Command& c, const CCommandQueue& queue) const
+{
+	int firstIndex = 0;
+	int lastIndex = queue.size() - 1;
+
+	if (c.GetOpts() & ALT_KEY) {
+		if (c.GetNumParams() >= 1)
+			firstIndex = std::max<int>(c.GetParam(0)-1, firstIndex);
+		if (c.GetNumParams() >= 2)
+			lastIndex = std::min<int>(c.GetParam(1)-1, lastIndex);
+	} else if (c.GetNumParams() > 0) {
```

Comment:
Slightly inconsistent where all other checks in this function use `>= (N+1)`  but here it's `> N`, consider
```suggestion
	} else if (c.GetNumParams() >= 1) {
```